modules/sitemaps/sitemaps.php

```diff
@@ -351,28 +319,43 @@ function jetpack_print_sitemap() {
 
 		// Image node specified in http://support.google.com/webmasters/bin/answer.py?hl=en&answer=178636
 		// These attachments were produced with batch SQL earlier in the script
-		if ( ! post_password_required( $post->ID ) && $attachments = wp_filter_object_list( $post_attachments, array( 'post_parent' => $post->ID ) ) ) {
-
-			$url['image:image'] = array();
+		if ( ! post_password_required( $post->ID ) ) {
+
+			$media = array();
+			$methods = array(
+				'from_thumbnail'  => false,
+				'from_slideshow'  => false,
+				'from_gallery'    => false,
+				'from_attachment' => false,
+				'from_html'       => false,
+			);
+			foreach ( $methods as $method => $value ) {
+				$methods[ $method ] = true;
+				$images_collected = Jetpack_PostImages::get_images( $post->ID, $methods );
```

Comment:
It's exactly for that: to fetch a different image or set of images each time. Check that in first iteration, the `from_thumbnail` item is set to true, the rest is false. After the image is fetch, the item is set to false again. In the second iteration, only `from_slideshow` is set to true. And so on. In this way, we collect all the images that `get_images`method can fetch.